java/sidecar/src/main/java/com/instaclustr/cassandra/sidecar/resource/OperationsResource.java

```diff
@@ -1,32 +1,90 @@
 package com.instaclustr.cassandra.sidecar.resource;
 
-import jmx.org.apache.cassandra.service.StorageServiceMBean;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 
 import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Optional;
+import java.util.UUID;
 
-@Path("/operations")
+import com.instaclustr.cassandra.sidecar.model.operation.BackupOperation;
+import com.instaclustr.cassandra.sidecar.model.operation.DecommissionOperation;
+import com.instaclustr.cassandra.sidecar.model.operation.Operation;
+import com.instaclustr.cassandra.sidecar.model.result.OperationResult;
+import com.instaclustr.cassandra.sidecar.operation.OperationExecutor;
+import com.instaclustr.cassandra.sidecar.operation.OperationTask;
+import com.instaclustr.cassandra.sidecar.operation.task.TaskFactory;
+
+@Path(OperationsResource.ENDPOINT)
 @Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
 public class OperationsResource {
-    private final StorageServiceMBean storageServiceMBean;
+
+    public static final String ENDPOINT = "/operations";
+
+    private final OperationExecutor operationExecutor;
+
+    private final TaskFactory taskFactory;
 
     @Inject
-    public OperationsResource(final StorageServiceMBean storageServiceMBean) {
-        this.storageServiceMBean = storageServiceMBean;
+    public OperationsResource(final OperationExecutor operationExecutor, final TaskFactory taskFactory) {
+        this.operationExecutor = operationExecutor;
+        this.taskFactory = taskFactory;
+    }
+
+    @GET
+    public Response getOperations() {
+        return Response.ok(operationExecutor.getOperations().stream().map(OperationTask::getOperationResult)).build();
+    }
+
+    @GET
+    @Path("{id}")
+    public Response getOperation(final @PathParam("id") UUID operationId) {
+
+        final Optional<OperationTask> operationTask = operationExecutor.getOperation(operationId);
+
+        if (operationTask.isPresent()) {
+            return Response.ok(operationTask.get().getOperationResult()).build();
+        } else {
+            return Response.status(NOT_FOUND).build();
+        }
     }
 
     @POST
-    @Path("/decommission")
-    public void decommissionNode() {
-        new Thread(() -> {
-            try {
-                storageServiceMBean.decommission();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }, "Decommission Background Thread").start();
+    public Response executeOperation(final Operation operation) {
+
+        if (operation.getId() == null) {
```

Comment:
The ID should never be provided by the client -- always generated by the server.